southworks/Recognizers-Text
# [Node JS] Fix test execution for Node JS versions > 10.8.0 (Heap out of memory)

## Proposed Changes
**Separate tests execution for each module to prevent Allocation Fail Error**.

Using Node JS versions higher than 10.8.0 was throwing an **Allocation Fail Error** running the tests due to a bump in the V8 version from 6.7 to 6.8 (introducing a change in heap memory limit).

Because of this, we separated the execution of the tests for each module in different scopes; and doing so, using `npm run test`, **AVA** executes the tests from different index files.

![image](https://user-images.githubusercontent.com/11904023/50290000-20261080-0449-11e9-9b75-8491e8a5681a.png)

## Testing
1. Use a Node JS version > 10.8.0.
2. Go to JavaScript directory.
3. Execute `npm install`.
4. Execute `npm run build`.
5. Execute `npm run test`.
6. Check that all test passed.

## Evidence
A comparation was made before and after the modifications for NodeJS version 10.9.0 and the last stable version at this moment 11.5.0.

### v10.9.0
**Before**
![image](https://user-images.githubusercontent.com/11904023/50300792-473e0b80-0464-11e9-9ce0-f37957303073.png)

**After**
![image](https://user-images.githubusercontent.com/11904023/50300809-51f8a080-0464-11e9-95ef-094707c77122.png)

### v11.5.0
**Before**
![image](https://user-images.githubusercontent.com/11904023/50300814-56bd5480-0464-11e9-81a2-fc7be5551fe6.png)

**After**
![image](https://user-images.githubusercontent.com/11904023/50300821-5c1a9f00-0464-11e9-85ff-bb4397bcac20.png)
